### ISSUE TITLE
Using lasso with stylus eventually gets confused, so an object looks selected, but you can't cut it.

### ISSUE THREAD

**Issue by frohro** [xournalpp/xournalpp#1993]
I have two HP 2760p laptops that I'm using for remote teaching as whiteboards.  The both need either
`$ xsetwacom set 13 TabletPCButton off`
or in the input preferences, "Merge button events with stylus tip events."  I choose the button as the lasso.  When I first login for a while, everything works as expected.  Then for some as yet unknown reason, I can lasso an object, and it shows as selected, but if I try and cut it, it sticks around.  If I move it slightly before cutting it is deleted.  I have seen it go from this state to working again, though only once.  It can be half an hour or more before it happens.  If I select the lasso by clicking the lasso tool it works just fine.  On my other laptop (a Lenovo Yoga 260) I don't have to select either way of merging events, and there is no problem.

Sorry I can't provide a way to trigger the behavior.  It just happens, but it always eventually happens, and it is the same with both 2760p's I have here.

**Comment by LittleHuba** [comment]
Please log such a session with the input log (description in the wiki). Also start a clock when you start the session so you know when this happened approximately. With the log we might be able to get to the source of this.

**Comment by frohro** [comment]
Thanks for the suggestion.  I went looking through the wiki and can't find the instructions for logging.  Can you provide a link or a little how to?
Thanks,
Rob

**Comment by rolandlo** [comment]
I think @LittleHuba refers to this [wiki page](https://github.com/xournalpp/xournalpp/wiki/How-to-provide-debug-info-for-input-related-bugs)

In the wiki you need to unfold "Pages" (above "Usage") to see some otherwise hidden wiki pages like the one above.

**Comment by LittleHuba** [comment]
Yes that's the one. It should also be linked in the template for any new input related bug.

**Comment by frohro** [comment]
Sorry I took so long to get this.  Here are the results.  I should mention that I did this on @rolandlo's branch for snapping to grid.  It has the same problem.

When I first reboot, everything works as expected for a while.  Then some unknown thing causes issues.  This time I gave up trying to cause it, but left xournalpp running.  Then after a while I came back, and it was not working.  The thing I do to test is to draw something then using the lasso tool activated from the pen button, select it, and try to cut it.  If it actually cuts, everything is working.  If it doesn't, its broken.  When I came back and tried, it was broken.  Then I did the gtk test.  I can reboot and do the gtk test starting from when it works and going until I notice it is not.
Thanks for trying to solve this for me!
[debug_gdk_input.log](https://github.com/xournalpp/xournalpp/files/4725962/debug_gdk_input.log)
[debug_input.log](https://github.com/xournalpp/xournalpp/files/4725963/debug_input.log)
[xinput_list.log](https://github.com/xournalpp/xournalpp/files/4725969/xinput_list.log)




Rob

**Comment by bhennion** [comment]
Closing as stale
